src/checkout-buttons/strategies/amazon-maxo/amazon-maxo-button-strategy.spec.ts

```diff
@@ -0,0 +1,171 @@
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.createButton).toHaveBeenCalled();
+            });
+
+            it('fails to create button if not PaymentMethod is supplied', async () => {
+                jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(undefined);
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions();
+
+                await expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(MissingDataError);
+            });
+
+            it('fails to create button if config is not initialized', async () => {
+                jest.spyOn(store.getState().config, 'getStoreConfig').mockReturnValue(undefined);
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions();
+
+                await expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(MissingDataError);
+            });
+
+            it('fails to create button if merchantId is not supplied', async () => {
+                jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(getPaymentMethodMockUndefinedMerchant());
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions();
+
+                await expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(InvalidArgumentError);
+            });
+
+            it('Validates if strategy has been initialized by spying on payment Processor initialize method', async () => {
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions();
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.initialize).toHaveBeenCalledTimes(1);
+            });
+
+            it('fails to initialize the strategy if not container id is supplied', async () => {
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions(Mode.UndefinedContainer);
+
+                await expect(() => strategy.initialize(checkoutButtonOptions)).toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if not a valid container id is supplied', async () => {
+                checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions(Mode.InvalidContainer);
+
+                await expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(InvalidArgumentError);
+            });
+
+        });
+    });
+
+    describe('#deinitialize()', () => {
+        let checkoutButtonOptions: CheckoutButtonInitializeOptions;
+
+        beforeEach(() => {
+            checkoutButtonOptions = getAmazonMaxoCheckoutButtonOptions(Mode.Full);
+            container = document.createElement('div');
+        });
+
+        afterEach(() => {
+            document.body.appendChild(container);
+        });
+
+        it('succesfully deinitializes the strategy', async () => {
```

Comment:
maybe add a test for calling `deinitializa` without calling `initialize` (expect to the exception is being thrown)

Reply:
it was added a test, when deinitalize has been called without called initialize.
deinitialize returns a promise with undefined.